dask/array/core.py

```diff
@@ -930,12 +937,23 @@ def store(sources, targets, lock=True, regions=None, compute=True,
     ))
 
     if return_stored:
+        load_dsks_mrg = sharedict.merge(*load_dsks)
+
         if compute:
             store_dlyds = [Delayed(k, store_dsks_mrg) for k in store_keys]
             store_dlyds = persist(*store_dlyds)
             store_dsks_mrg = sharedict.merge(*[e.dask for e in store_dlyds])
 
-        load_dsks_mrg = sharedict.merge(store_dsks_mrg, *load_dsks)
+        load_dsks_mrg = sharedict.merge(load_dsks_mrg, store_dsks_mrg)
+
+        if compute:
+            from ..optimization import cull, inline
+            load_dsks_mrg = cull(inline(load_dsks_mrg), load_keys)[0]
+        else:
+            from ..optimization import inline_functions
```

Comment:
Trying to import either of these outside the function results in an `ImportError` during testing. So have imported them locally only where they are needed to avoid this issue.